internal/k8s_resource.go

```diff
@@ -0,0 +1,356 @@
+package provider
```

Comment:
can we move parsing logic and their structs (leave args ans state in this file) to a new file called `k8s_parser`